service/matching/matching_engine.go

```diff
@@ -864,6 +865,22 @@ func (e *matchingEngineImpl) QueryWorkflow(
 			return nil, serviceerror.NewInternal("unknown query completed type")
 		}
 	case <-ctx.Done():
+		// task timed out. log (optionally) and return the timeout error
+		ns, err := e.namespaceRegistry.GetNamespaceByID(namespaceID)
```

Comment:
Instead of calling namespace registry, constructing log tags, etc. yet again, we have a `tqm` here, which already has a tagged logger and a `taskQueueConfig` with tq-specific configs. Can we somehow delegate this logging to that, or do this inside of that?

Reply:
I couldn't find a tagged logger in taskQueueManager implementation [here](https://github.com/temporalio/temporal/blob/chetan_query_workflow_logging/service/matching/task_queue_manager.go#L162-L195). There is a `taggedMetricsHandler` inside it but no tagged logger. Did I miss something?

Reply:
That logger is tagged, see https://github.com/temporalio/temporal/blob/chetan_query_workflow_logging/service/matching/task_queue_manager.go#L235-L238
(There's no separate type for it so we just call it a logger. Arguably the metricsHandler should just be called that and not have "tagged" in the name either.)

Reply:
Got it. I'll see how I can log using `tqm`. It's often not a good idea to reach into internals of other objects. So, I'll probably make the matcher's logger be auto tagged as well if it's easier.
As far as logging location, this seems like the right location since we are handling the context deadline and is at a reasonably high level to log the data.

Reply:
Yeah, I see that it may be awkward to do, I was just noticing the duplication and want to see if there's a way to refactor it. E.g. possibly moving everything after the call to `tqm.DispatchQueryTask` inside of that function or in another function on the tqm (including the query map). That's a larger change and may be bad for other reasons, but worth thinking about.

Reply:
Yes. Looks like we can't have a tagged logger in matcher since its internal state doesn't have the namespace, queue name/type etc like `tqm`. So a better approach is to rethink this whole code block and see if it belongs inside TaskQueueManager.
Even that doesn't seem like the right place to me since the channel is created here and attached to `e.queryResults` and eventually read here. So a refactor would lead to a larger change beyond the scope of this PR. I do agree that this file is littered with lot of log tagging and requires some form of cleanup.

Reply:
Resolving this to unblock the PR.